### PR TITLE
removes references to hardcoding, regularizes whitelist note

### PR DESCRIPTION
**Pull request by acozine**
##### SUMMARY
Following up on a discussion with @bcoca, this PR removes the last references in the plugin docs to things that used to be hardcoded. 

In the interests of tidiness, it also makes all the references in the plugin docs to "whitelisting" the same.

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs.ansible.com

### CODE REVIEW

File: lib/ansible/plugins/callback/tree.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
     callback: tree
     callback_type: notification
     requirements:
+      - whitelist in configuration
```
Comment (bcoca) [inline lib/ansible/plugins/callback/tree.py:12]: this is used even w/o whitelisting when the `--tree` option is used, so not always a requirement

File: lib/ansible/plugins/inventory/ini.py [hunk 2]
```diff
@@ -23,9 +23,9 @@
           Unlike host lines, C(:vars) sections accept only a single entry per line, so everything after the C(=) must be the value for the entry.
         - Do not rely on types set during definition, always make sure you specify type with a filter when needed when consuming the variable.
         - See the Examples for proper quoting to prevent changes to variable type.
+    requirements:
+        - whitelist in configuration
```
Comment (bcoca) [inline lib/ansible/plugins/inventory/ini.py:27]: this one is whitelisted by default

**Review by bcoca — CHANGES_REQUESTED** [review]

**Comment by dagwieers** [comment]
Personally I think the requirement "whitelist in configuration" is very confusing.

Can we not make this a bit more explicit, or add a section to the documentation and a reference to it instead ?

Also some of the entries are fully lowercase, which I don't like either, but the contents of the requirements has been a garbage bin in the past. It is being used for various things in various formats. A bit of consistency would be nice, possibly reorganizing that data-structure.

Personally I would rather make this part of the notes, and even add a "seealso" entry pointing to a documentation section describing how to enable plugins. And keep the requirements for software/hardware/version listings.

**Comment by acozine** [comment]
Closing in favor of #49383.

**Comment by acozine** [comment]
Thanks @dagwieers @bcoca I'll keep the change smaller for now and we can continue to chip away at improving the plugin docs.

**Pull request closed without merging**